src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,182 @@
+package duke;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import static java.lang.Integer.parseInt;
+
+/**
+ * The Parser class processes the data inputs from the Ui, and uses this input to alter the list
+ */
+public class Parser {
+
+    /**
+     * Checks the date input from the user to see if it is a valid date
+     *
+     * @param dateStr Date taken in to check for validity
+     * @return If date is valid
+     */
+    public static boolean isValid(String dateStr) {
+        try {
+            LocalDate.parse(dateStr, DateTimeFormatter.ISO_LOCAL_DATE);
+        } catch (DateTimeParseException e) {
+            System.out.println("Invalid date format. Please enter in the format: yyyy-mm-dd");
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * The main method that processes the Ui input,
+     * @param in The input from the Ui to be processed
+     * @param ui Ui to close when necessary
+     * @param tasklist The list that is adjusted when a specific input is given
+     */
+    public static void parse(String in, Ui ui, TaskList tasklist) {
```

Comment:
Should a parser be responsible in performing commands or should it just parse user input such that it can be meaningful to the program?